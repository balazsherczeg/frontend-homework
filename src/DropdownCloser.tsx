import { connect } from 'react-redux'

import { hideDropdownMenu } from './actions'

type OwnProps = {
  children: React.ReactNode,
}

type DispatchProps = {
  hideDropdownMenu: () => void,
}

export const DropdownCloser = ({ children, hideDropdownMenu }: OwnProps & DispatchProps) => (
  <div
    className="DropdownCloser"
    onClick={ hideDropdownMenu }
  >
    {children}
  </div>
)

export default connect(
  null,
  { hideDropdownMenu }
)(DropdownCloser)
