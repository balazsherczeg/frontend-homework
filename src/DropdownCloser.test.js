import { shallow } from 'enzyme'
import { DropdownCloser } from './DropdownCloser'

describe('<DropdownCloser />', () => {
  const hideDropdownMenuMock = jest.fn()

  const render = () => shallow(
    <DropdownCloser
      hideDropdownMenu={hideDropdownMenuMock}
    >
      it is a child
    </DropdownCloser>
  )

  it('renders with a child inside', () => {
    expect(render()).toMatchSnapshot()
  })

  it('calls hideDropdownMenu when clicked', () => {
    const component = render()
    component.simulate('click')
    expect(hideDropdownMenuMock).toHaveBeenCalled()
  })
})
