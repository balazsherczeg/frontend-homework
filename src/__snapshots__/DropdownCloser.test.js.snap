// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownCloser /> renders with a child inside 1`] = `
<div
  className="DropdownCloser"
  onClick={[MockFunction]}
>
  it is a child
</div>
`;
